imap_processing/io.py

```diff
@@ -0,0 +1,162 @@
+"""Input/output capabilities for the IMAP data processing pipeline."""
```

Comment:
We will use this file to make API call from within batch job, right? Looks good to me.

Reply:
yeah, that sounds like a good plan.